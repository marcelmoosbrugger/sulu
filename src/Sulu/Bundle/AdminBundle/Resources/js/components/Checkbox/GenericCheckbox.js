// @flow
import React from 'react';
import classNames from 'classnames';
import Icon from '../Icon';
import type {CheckboxProps} from './types';
import genericCheckboxStyles from './genericCheckbox.scss';

type Props = CheckboxProps & {
    className: string,
    icon?: string,
};

export default class GenericCheckbox extends React.PureComponent<Props> {
    static defaultProps = {
        className: '',
    };

    handleChange = (event: SyntheticEvent<HTMLInputElement>) => {
        if (this.props.onChange) {
            this.props.onChange(event.currentTarget.checked, this.props.value);
        }
    };

    render() {
        const checkboxClass = classNames({
            [genericCheckboxStyles.checkbox]: true,
            [this.props.className]: !!this.props.className,
        });

        return (
            <label className={genericCheckboxStyles.genericCheckbox}>
                <span className={checkboxClass}>
                    <input
                        type="checkbox"
                        name={this.props.name}
                        checked={this.props.checked}
                        onChange={this.handleChange} />
                    <span>
                        {this.props.icon && <Icon name={this.props.icon} />}
                    </span>
                </span>
                {this.props.children && <span>{this.props.children}</span>}
            </label>
        );
    }
}
