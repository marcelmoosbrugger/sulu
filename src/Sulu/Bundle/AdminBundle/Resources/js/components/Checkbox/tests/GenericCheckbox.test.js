/* eslint-disable flowtype/require-valid-file-annotation */
import {shallow, render} from 'enzyme';
import React from 'react';
import GenericCheckbox from '../GenericCheckbox';

test('The component should render in unchecked state', () => {
    const checkbox = render(<GenericCheckbox checked={false} />);
    expect(checkbox).toMatchSnapshot();
});

test('The component should render in checked state', () => {
    const checkbox = render(<GenericCheckbox checked={true} />);
    expect(checkbox).toMatchSnapshot();
});

test('The component should render in with class', () => {
    const checkbox = render(<GenericCheckbox className="my-class" checked={false} />);
    expect(checkbox).toMatchSnapshot();
});

test('A click on the checkbox should trigger the change callback', () => {
    const onChangeSpy = jest.fn();
    const checkbox = shallow(<GenericCheckbox checked={false} onChange={onChangeSpy} />);
    checkbox.find('input').simulate('change', {currentTarget: {checked: true}});
    expect(onChangeSpy).toHaveBeenCalledWith(true, undefined);
    checkbox.find('input').simulate('change', {currentTarget: {checked: false}});
    expect(onChangeSpy).toHaveBeenCalledWith(false, undefined);
});

test('A click on the checkbox should trigger the change callback with the value', () => {
    const onChangeSpy = jest.fn();
    const checkbox = shallow(<GenericCheckbox checked={false} value="my-value" onChange={onChangeSpy} />);
    checkbox.find('input').simulate('change', {currentTarget: {checked: true}});
    expect(onChangeSpy).toHaveBeenCalledWith(true, 'my-value');
    checkbox.find('input').simulate('change', {currentTarget: {checked: false}});
    expect(onChangeSpy).toHaveBeenCalledWith(false, 'my-value');
});
