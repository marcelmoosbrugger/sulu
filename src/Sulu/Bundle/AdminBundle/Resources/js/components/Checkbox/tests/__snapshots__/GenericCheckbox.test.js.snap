// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render in checked state 1`] = `
<label
  class="genericCheckbox"
>
  <span
    class="checkbox"
  >
    <input
      checked=""
      type="checkbox"
    />
    <span />
  </span>
</label>
`;

exports[`The component should render in unchecked state 1`] = `
<label
  class="genericCheckbox"
>
  <span
    class="checkbox"
  >
    <input
      type="checkbox"
    />
    <span />
  </span>
</label>
`;

exports[`The component should render in with class 1`] = `
<label
  class="genericCheckbox"
>
  <span
    class="checkbox my-class"
  >
    <input
      type="checkbox"
    />
    <span />
  </span>
</label>
`;
